crates/sui/src/client_ptb/ast.rs

```diff
@@ -141,6 +140,27 @@ pub enum ParsedPTBCommand {
     Preview,
 }
 
+#[derive(Debug, Clone)]
+pub struct ParsedStructType {
+    pub fq_name: ModuleAccess,
+    pub type_args: Vec<ParsedType>,
+}
+
+#[derive(Debug, Clone)]
+pub enum ParsedType {
```

Comment:
The typetag parser no longer does deferred resolution of named addresses. So this needed to be moved into here so that we could support substitution of named addresses based on the current environment when building a PTB.

Reply:
Why not? This was the whole point of having `ParsedType` vs `TypeTag` or any similar form

Reply:
The main reason is a balk at exposing `ParsedType` from `move-core-types`, and if we want to keep `ParsedType` while not having it in `move-core-types` we either need to make that parser the source of truth, and the parser should then be in a different crate than `move-core-types`. However then you can't write `from_str` instances for it and there's a generally fairly strong feeling that the parser for type tags should live in `move-core-types`, which I think is a fair point. 

Generally, simplifying this form so that we resolve while parsing feels simpler to me as we don't need to then build up a bunch of parallel datastructures to `TypeTag` to handle unresolved addresses in types. Additionally, the existing `ParsedType` form was only used for testing code (which can be handled by deferring the parsing of the type until the usage site in the tests), or in the PTB cli case but there we're already parsing into our own form with our own parser anyways so we can capture source info, so there isn't any really advantage to keeping it for that.

Reply:
> The main reason is I'm not a gigantic fan of exposing ParsedType from move-core-types

Why? If you are exposing the parser, whats the harm? 
You can have the `from_str` instances of `TypeTag` operate over an empty mapping of named addresses, it will still then round trip from `Display`

Reply:
I guess to me it's just one more thing that you then need to worry about, and adds unneeded complexity, both to the parser and data structures (now you need to worry about address format, how the different parsed data structures matchup with the type tag, etc), as well as the interface of the crate. 

I would prefer to keep things as tight as possible around type tags and the interface around them, as in the past I think we've offered a bit too much flexibility and different options in this area. Basically, I want to move us into a world where there is only one type of "type" or parsed type as far as move core types is concerned, and that is `TypeTag`. You can go from a string into a `TypeTag`, and you can resolve named addresses if you want while doing that, but that's it, and once you have a `TypeTag` that's it.

Reply:
Might be easier to chat about this synchronously, so let's discuss at the open discussion tomorrow if we have time.

Reply:
Imo named addresses and type tag parsing should likely be handled separately.